code/client/munkilib/updatecheck.py

```diff
@@ -2738,11 +2739,19 @@ def download_icons(item_list):
     munkicommon.display_debug2('Icon base URL is: %s', icon_base_url)
     for item in item_list:
         icon_name = item.get('icon_name') or item['name']
+        pkginfo_icon_hash = item.get('icon_hash')
         if not os.path.splitext(icon_name)[1] in icon_known_exts:
             icon_name += '.png'
         icon_list.append(icon_name)
         icon_url = icon_base_url + urllib2.quote(icon_name)
         icon_path = os.path.join(icon_dir, icon_name)
+        if os.path.isfile(icon_path):
+            xattr_hash = fetch.getxattr(icon_path, XATTR_SHA)
+            if not xattr_hash:
+              xattr_hash = munkicommon.getsha256hash(icon_path)
+              fetch.writeCachedChecksum(icon_path, XATTR_SHA)
```

Comment:
Yeah, I was trying to do some last minute work on this as I rushed out the door, which is always a mistake.